src/main/java/roomescape/time/dao/TimeDao.java

```diff
@@ -0,0 +1,69 @@
+package roomescape.time.dao;
+
+import java.time.LocalTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import org.springframework.stereotype.Component;
+import roomescape.reservation.exception.InvalidReservationRequestException;
+import roomescape.time.model.Time;
+
+@Component
+public class TimeDao {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final SimpleJdbcInsert insert;
+
+    public TimeDao(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.insert = new SimpleJdbcInsert(jdbcTemplate)
+            .withTableName("time")
+            .usingGeneratedKeyColumns("id");
+    }
+
+    public List<Time> findAll() {
+        return jdbcTemplate.query(
+            "select id, time from time",
+            (resultSet, rowNum) ->
+                new Time(
+                    resultSet.getLong("id"),
+                    resultSet.getTime("time").toLocalTime()
+                )
+        );
+    }
+
+    public Optional<Time> findById(final Long id) {
+        Time time;
+        try {
+            time = jdbcTemplate.queryForObject(
+                "select id, time from time where id = ?",
+                (resultSet, rowNum) -> new Time(
+                    resultSet.getLong("id"),
+                    resultSet.getTime("time").toLocalTime()
+                ), id
+            );
+        } catch (NumberFormatException | DataAccessException e) {
+            throw new InvalidReservationRequestException(
+                "Invalid time ID: " + id);
+        }
+
+        return Optional.ofNullable(time);
+    }
```

Comment:
반환값이 `Optional`인데, catch에서 예외를 발생시킬 필요가 있을까요?

또한 `NumberFormatException`은 왜 발생하는 예외인가요?

```suggestion
        try {
            Time time = jdbcTemplate.queryForObject(
                "select id, time from time where id = ?",
                (resultSet, rowNum) -> new Time(
                    resultSet.getLong("id"),
                    resultSet.getTime("time").toLocalTime()
                ), id
            );
            return Optional.ofNullable(time);
        } catch (DataAccessException e) {
            return Optional.empty();
        }
```

Reply:
`Optional.ofNullable`을 사용했다는 건 비어있을 수도 있는 값이라는 것인데.. 예외를 발생시킬 필요가 없네요.. 의도가 모호한 코드를  작성했습니다 ..! 

`NumberFormatException`는 문자열을 숫자로 변환하려고할 때 발생하는 예외인데, 
지금 id가 Long 타입으로 지정되어있어 발생할 일이 없는데 불필요하게 작성했었네요. 모두 수정했습니다👍